### PR TITLE
Adding AlexNet

### PR DESCRIPTION
**Pull request by kartikdutt18**
Hi everyone, I was working on adding alexnet to models however mlpack currently doesn't have adaptive pooling implemented. So I stopped here, the following things are left to be done:

- [x] Add AlexNet.
- [x] Write cmake to include alexnet as `#include<models/alexnet>`
- [ ] Addition of pre-trained weights.
- [ ] Changes in readme to show how models such as this one can be trained on their dataset.

**Currently Working on writing Cmake to resolve task 2.**

I hope this is ok.
Thanks.

### CODE REVIEW

**Comment by kartikdutt18** [comment]
Sorry for the delay, I will complete everything with a done or two.

**Comment by kartikdutt18** [comment]
Hi @zoq, I am terribly sorry to disturb for a trivial task but could you have a look at Cmake I wrote. I would appreciate any help or suggestions. Thanks.

File: models/alexnet.hpp [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+   * @param kernelHeight Height of the filter/kernel.
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   */
+  Sequential<>* PoolingBlock(const size_t kernelWidth,
+                             const size_t kernelHeight,
+                             const size_t strideWidth = 1,
+                             const size_t strideHeight = 1)
+  {
+    Sequential<>* poolingBlock;
```
Comment (kartikdutt18) [inline models/alexnet.hpp]: Error Here, Will fix it in the next commit.

**Comment by kartikdutt18** [comment]
Another nice idea for this would be to specify which model to be used on the dataset using mlpack CLI.

**Comment by prince776** [comment]
Hi @kartikdutt18 I'm not sure if you're still working on this model, but AlexNet also had Local Response Normalization Layer, and I didn't see it in the code. Was this also something you were working on?

**Comment by kartikdutt18** [comment]
I haven't added LRN layer, I have skipped it for now. This is alexnet without that layer since it wasn't the main purpose of alexnet. I know that you are working on it so whenever it gets merged I'll add it here as well but as you know restructuring is taking most of my time. I plan on merging this after restructuring. Till then I am keeping it as it is.

**Comment by kartikdutt18** [comment]
closing this, Will be moved to the new repo.

**Pull request closed without merging**